apis/r/R/SOMADataFrame.R

```diff
@@ -178,7 +178,7 @@ SOMADataFrame <- R6::R6Class(
     #' @param iterated Option boolean indicated whether data is read in call (when
     #' `FALSE`, the default value) or in several iterated steps.
     #' @param log_level Optional logging level with default value of `"warn"`.
-    #' @return An [`arrow::Table`].
+    #' @return arrow::\link[arrow]{Table} or \link{TableReadIter}
```

Comment:
As above: let's simply have this always iterate. Then `concat` is an easy tack-on to the return value from this.